build.rs

```diff
@@ -1,417 +1,182 @@
-                .unwrap();
         }
-
-        run_bindgen(
-            &PathBuf::from(env::var("OUT_DIR").unwrap()).join(BINDINGS_FILE),
-            Some(&PathBuf::from(include)),
-        ).expect("russimp generate bindgen failed, for details see https://github.com/jkvargas/russimp-sys");
-        println!("cargo:rustc-link-search=native={}", libdir);
-        println!("cargo:rustc-link-lib={}", libname);
     }
-}
 
-fn assimp_dynamic_linking() -> (String, String, String) {
-    let target = std::env::var("TARGET").unwrap();
-    let vcpkg_root = std::env::var("VCPKG_ROOT").unwrap_or("target/vcpkg".to_string());
+    let file = fs::File::open(out_dir.join(&archive_name)).unwrap();
+    let mut archive = tar::Archive::new(GzDecoder::new(file));
+    archive.unpack(out_dir).unwrap();
+}
 
-    let include_path = if target.contains("apple") {
-        "/opt/homebrew/opt/assimp/include"
-    } else if std::env::var("DOCS_RS").is_ok() {
-        "assimp/include"
-    } else {
-        "/usr/local/include"
-    };
+fn main() {
+    println!("cargo:rerun-if-changed=bin/package/main.rs");
 
-    let mut lib = vcpkg::Config::new()
-        .vcpkg_root(vcpkg_root.into())
-        .find_package("assimp")
-        .unwrap_or(vcpkg::Library {
-            include_paths: vec![PathBuf::from(include_path)],
-            link_paths: vec![PathBuf::from("/usr/local/lib")],
-            found_names: vec!["assimp".to_owned()],
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
 
-            ports: vec![],
-            cargo_metadata: vec![],
-            dll_paths: vec![],
-            found_dlls: vec![],
+    if cfg!(feature = "build-assimp") {
+        build_from_source();
+    } else if cfg!(feature = "prebuilt") {
+        link_from_package();
+    }
 
-            is_static: false,
-            found_libs: vec![],
-            vcpkg_triplet: "".to_string(),
-        });
+    bindgen::builder()
+        .header("wrapper.h")
+        .clang_arg(format!("-I{}", out_dir.join(static_lib()).join("include").display()))
+        .clang_arg(format!("-I{}", "assimp/include"))
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        .allowlist_type("ai.*")
+        .allowlist_function("ai.*")
+        .allowlist_var("ai.*")
+        .allowlist_var("AI_.*")
+        .derive_partialeq(true)
+        .derive_eq(true)
+        .derive_hash(true)
+        .derive_debug(true)
+        .generate()
+        .unwrap()
+        .write_to_file(out_dir.join("bindings.rs"))
+        .expect("Could not generate russimp bindings, for details see https://github.com/jkvargas/russimp-sys");
+
+    let mut built_opts = built::Options::default();
+    built_opts
+        .set_dependencies(false)
+        .set_compiler(false)
+        .set_ci(false)
+        .set_cfg(false);
+    built::write_built_file_with_opts(&built_opts, &manifest_dir, &out_dir.join("built.rs"))
```

Comment:
I am using `built` to have a target constant for my local package generation since TARGET isn't available outside of  `build.rs`. It simply generates constants related to the platform the crate was built on.

That's the best solution I could come up with.